lib/ykpiv.c

```diff
@@ -154,40 +184,15 @@ ykpiv_rc ykpiv_connect(ykpiv_state *state, const char *wanted) {
     }
     rc = SCardConnect(state->context, reader_ptr, SCARD_SHARE_SHARED,
         SCARD_PROTOCOL_T1, &state->card, &active_protocol);
-    if(rc != SCARD_S_SUCCESS)
-    {
+    if(rc != SCARD_S_SUCCESS) {
       if(state->verbose) {
         fprintf(stderr, "SCardConnect failed, rc=%08lx\n", rc);
       }
       continue;
     }
-
-    {
-      APDU apdu;
-      unsigned char data[0xff];
-      unsigned long recv_len = sizeof(data);
-      int sw;
-      ykpiv_rc res;
-
-      memset(apdu.raw, 0, sizeof(apdu));
-      apdu.st.ins = 0xa4;
-      apdu.st.p1 = 0x04;
-      apdu.st.lc = sizeof(aid);
-      memcpy(apdu.st.data, aid, sizeof(aid));
-
-      if((res = send_data(state, &apdu, data, &recv_len, &sw)) != YKPIV_OK) {
-        if(state->verbose) {
-          fprintf(stderr, "Failed communicating with card: '%s'\n", ykpiv_strerror(res));
-        }
-        continue;
-      } else if(sw == SW_SUCCESS) {
-        return YKPIV_OK;
-      } else {
-        if(state->verbose) {
-          fprintf(stderr, "Failed selecting application: %04x\n", sw);
-        }
-      }
-    }
+    if (_select_application(state) != YKPIV_OK)
```

Comment:
Please add {} (as previously noted we've been inconsistent about this but want them present.